### PR TITLE
Fixing python unit tests

### PR DESCRIPTION
**Pull request by alanch-ms**
Apologies for the massive changes.  I tried to group the changes into logical self-contained commits.  I would recommend reviewing them by individual commits instead of everything as a whole.  

Note: I would definitely review it using [ignore space](http://www.codersrevolution.com/blog/How-To-Ignore-Whitespace-in-GitHub-Unified-Diff) setting.  

Also please take a look at the doc [README_uwp.md](https://github.com/alanch-ms/python/blob/unittest/README_uwp.md)

Here are the results after this change.  Note: `test_ctypes` is failing because `ctypes`'s callback are not ported yet.  Since we might want to port this functionality in the future, I didn't disable those tests.  Currently, they will throw a MemoryError since it cannot allocate a RXW memory page.

334 tests OK.
1 test failed:
    test_ctypes
2 tests altered the execution environment:
    test_site test_warnings
59 tests skipped:
    test_asdl_parser test_cmd_line test_crypt test_curses test_dbm_gnu
    test_dbm_ndbm test_devpoll test_distutils test_epoll test_fcntl
    test_fork1 test_gdb test_grp test_idle test_ioctl test_kqueue
    test_lzma test_mmap test_msilib test_multiprocessing_fork
    test_multiprocessing_forkserver test_multiprocessing_main_handling
    test_multiprocessing_spawn test_nis test_openpty test_ossaudiodev
    test_pipes test_poll test_posix test_pty test_pwd test_readline
    test_resource test_smtpnet test_socketserver test_spwd test_sqlite
    test_startfile test_subprocess test_syslog test_tcl
    test_threadsignals test_timeout test_tix test_tk test_tracemalloc
    test_ttk_guionly test_ttk_textonly test_univnewlines
    test_urllib2net test_urllibnet test_venv test_wait3 test_wait4
    test_webbrowser test_winreg test_winsound test_xmlrpc_net
    test_zipfile64

### CODE REVIEW

**Comment by alanch-ms** [comment]
@munyirik @bfjelds  @alecont 
please CR when you have a chance

File: README_uwp.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
+# Python for UWP
+
+PythonUWP is based on a modified version of CPython 3.5 which supports running as a Windows 10 UWP (Universal Windows Platform) application.  Specifically, this supports running in a background application on Windows 10 IoT core.
```
Comment (munyirik) [inline README_uwp.md]: I would link "UWP" with https://msdn.microsoft.com/en-us/library/windows/apps/dn894631.aspx. Just for people who want to look into what it means in more detail.

**Comment by munyirik** [comment]
Made a few comments in the commits. Overall :+1:*

*With limited Python knowledge.

**Comment by alanch-ms** [comment]
Thanks Munyiri!  Anyone else?

File: cpython/Lib/concurrent/futures/__init__.py [hunk 2]
```diff
@@ -14,5 +14,11 @@
                                       Executor,
                                       wait,
                                       as_completed)
-from concurrent.futures.process import ProcessPoolExecutor
+try:
+    from concurrent.futures.process import ProcessPoolExecutor
+except ImportError:
+    import sys
+    if sys.platform != 'uwp':
```
Comment (kevin-brown) [inline cpython/Lib/concurrent/futures/__init__.py]: Would it make more sense to do this check _before_ attempting to import the module?

Comment (alecont) [inline cpython/Lib/concurrent/futures/__init__.py]: :+1:

**Comment by alecont** [comment]
looked over the changes! :-) me likey. Please do fix the issue reported by kevin, then we can merge.

File: cpython/Lib/concurrent/futures/__init__.py (hunk 2 above)
Comment (alanch-ms) [inline cpython/Lib/concurrent/futures/__init__.py]: Thanks for the feedback, Kevin!  I have updated my change.

**Pull request merged**